### PR TITLE
Symbols to get the date of the next occurrence of a specific day of the week

### PR DESCRIPTION
**Pull request by tcnh**
Discussed this one a while ago. Was still on my hard drive. See if you think it's useful

### CODE REVIEW

File: src/main/java/nl/hsac/fitnesse/symbols/NextXDay.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+
+/**
+ * Abstract class to get the next occurrence of a specific weekday. Can be called with +x to skip x occurrences
+ */
+
+public abstract class NextXDay extends Today {
+    private int DAY;
```
Comment (fhoeben) [inline src/main/java/nl/hsac/fitnesse/symbols/NextXDay.java]: Java standard for (non-static) fields is camel case (e.g. `day`). Is there a reason you chose capitals here?

**Comment by fhoeben** [comment]
I notice now that 'nextFriday' returns today's date if called on a Friday, is that intentional?
To me, the name 'next...' is a bit misleading if it might give today's date. What would you think of just '!Friday', or '!thisFriday'?

The increment indicates the number of weeks to add/subtract, correct?
While this is quite logical I'm thinking it might be more powerful if it just adds a number of days, allowing the user to say 'two days beyond this Friday'. Adding a multiple of weeks is just a matter of multiplying by 7 and one can express more that way.  What are your thoughts?

**Comment by tcnh** [comment]
I notice now that 'nextFriday' returns today's date if called on a Friday, is that intentional?
To me, the name 'next...' is a bit misleading if it might give today's date. What would you think of just '!Friday', or '!thisFriday'?
This was a discussion in the project as well. For us, the first occurrence of the date, not in the past sufficed, so we chose to include today if it happened to be the right day of the week. I agree on the 'next' being misleading in that case.

> The increment indicates the number of weeks to add/subtract, correct?
> While this is quite logical I'm thinking it might be more powerful if it just adds a number of days, allowing the user to say 'two days beyond this Friday'. Adding a multiple of weeks is just a matter of multiplying by 7 and one can express more that way. What are your thoughts?

Agree. Also less confusing compared to !today

**Comment by keetron** [comment]
My comment when I read the release notes: mention that !Monday will give you the date for the coming Monday relative to the date of the current test execution. Awesome addition tho!
Can I make an edit here https://github.com/fhoeben/hsac-fitnesse-plugin/wiki/1.-Symbol-Samples#relative-dates ?

**Comment by fhoeben** [comment]
An addition on that page would be appreciated.  

I was struggling how to give a concise description of the current date if the test execution is on a Monday or the coming Monday if it is not a Monday today.
'Coming' (like next) to me suggests that today's date will never be returned...

**Comment by keetron** [comment]
My bad, I misinterpreted !Monday to mean the very next occurance of Monday but after reviewing the code (at all ;-) ) I see from the very clean code and nice comments that today counts as well. I'll make the addition to the wiki and source.

**Comment by keetron** [comment]
Proposal done in wiki, not sure if I like it but there would be a lot of repetition.

**Comment by fhoeben** [comment]
Repetition is indeed an issue. And you have to make sure to have all texts reflect the right 'increment', i.e. 'Date for next Tuesday if today is not a Tuesday' should be 'Date for three days after next Tuesday if today is not a Tuesday'

Did you see the attempt I made (without explaining the usage of increment) in: https://raw.githubusercontent.com/fhoeben/hsac-fitnesse-fixtures/master/wiki/FitNesseRoot/HsacExamples/SymbolSamples.wiki

**Pull request merged**